marklogic-data-hub/src/test/java/com/marklogic/hub/flow/impl/FlowRunnerTest.java

```diff
@@ -247,6 +249,34 @@ public void testIngestCSVasXMLCustomDelimiter(){
         Assertions.assertEquals(count, 25);
     }
 
+    @Test
+    public void testSourceQueryLimit(){
+        Map<String,Object> opts = new HashMap<>();
+
+        runAsDataHubOperator();
+        RunFlowResponse resp = runFlow("testFlow", "4", UUID.randomUUID().toString(),opts, new HashMap<>());
+        flowRunner.awaitCompletion();
+        Assertions.assertTrue(getDocCount(HubConfig.DEFAULT_STAGING_NAME, "csv-tab-coll") == 25);
```

Comment:
Should use assertEquals(25, getDocCount) . Also, why does this first test use step 4, while the others use step 6? It's a stronger test if the same step is always used, as the first test shows the expected number with no limit applied.

Reply:
The first test that runs step 4 is an ingestion step that ingests 25 docs. Rest all run step 6 with limit applied. I will update the assertions. I just copied them from existing tests.